app/features/utils/functions.js

```diff
@@ -3,7 +3,34 @@
 // @flow
 
 import { shell } from 'electron';
-import config from '../config';
+
+/**
+ * Finds the external_api.js for the given Server URL. It also handles if the
+ * server URL contains tenant name.
+ *
+ * @param {string} serverURL - Jitsi Meet Server URL.
+ * @returns {Promise} - Promise which returns the external_api.js url.
+ */
+export async function findExternalApiURL(serverURL: string) {
+    // eslint-disable-next-line no-param-reassign
+    serverURL = normalizeServerURL(serverURL);
+    const apiURL = getExternalApiURL(serverURL);
+
+    const response = await fetch(apiURL, { method: 'HEAD' });
+
+    if (response.ok) {
+        return apiURL;
+    }
+
+    const start = 'https://'.length;
```

Comment:
I really don't like this. We are not leveraging the work URL objects already do for us. They are cheap, let's use them. Also I just tested it and browsers seem to update href when pathname is changed. Maybe we need to use the whatwg-url polyfill, that's ok.

I know this is tedious, and I'm sorry, but I want to get it right, and I think we are not there yet. If you want to leave this and come back to it later, that's ok too!

Reply:
> I know this is tedious, and I'm sorry, but I want to get it right,

Don't say sorry. I want to get it right too. :smile: 

> We are not leveraging the work URL objects already do for us. They are cheap, let's use them. Also I just tested it and browsers seem to update href when pathname is changed. Maybe we need to use the whatwg-url polyfill, that's ok.

Maybe I was doing something wrong. I'll try again. 


@saghul 
Let's do this. We'll revisit this after we fix #67 `Handle URL Parameters.` issue.
Maybe I will have more clarity about how to get it right as we'll need to parse complete url there.
:smile:

Reply:
Sounds good! Thanks man!